doc/testing_tools/testing_floating_points.qbk

```diff
@@ -135,29 +140,42 @@ inequations [link equ2 (2)] and [link equ3 (3)] where all underflow, overflow co
 || abs(u - v)/abs(v) <= epsilon; // (5)
 ``
 
+The above however, will not work when `v` or `u` is too close to zero. In such cases the solution is to resort to [link equ1 (1)]. 
+
 [h3 Implementation]
 
-Checks based on equations [link equ4 (4)] and [link equ5 (5)] is implemented by the binary predicate 
-[classref boost::math::fpc::close_at_tolerance `close_at_tolerance`]. The method for comparison (Equ [link equ4 (4)] or [link equ5 (5)]) as well as the
-epsilon are selected at construction. This predicate reports the tolerance to set to make the test successful.
+Binary predicate [classref boost::math::fpc::close_at_tolerance `close_at_tolerance`] implements checks [link equ4 (4)] and [link equ5 (5)].
+The tolerance and the seleciotn between [link equ4 (4)] and [link equ5 (5)] is determined during construction. Apart from the Boolean result,
+the predicate can also return the minimum tolerance required to check positively the closeness of `u` and `v`.
+
+Unary predicate [classref boost::math::fpc::small_with_tolerance `small_with_tolerance`] implements check [link equ1 (1)].
+
+Assertion __BOOST_TEST__ (when comparing floating-point numbers) uses the following algorithm:
+
+* When either `u` or `v` is zero, evaluates [classref boost::math::fpc::small_with_tolerance `small_with_tolerance`] on the other.
+* When the specified tolerance is zero, preforms direct (native) comparison between `u` and `v`.
+* Otherwise, performs [classref boost::math::fpc::close_at_tolerance `close_at_tolerance`] using [link equ4 (4)].
+
+[note Therefore in order to check if a number is close to zero with tolerance, you need to type: 
+```
+BOOST_TEST(v == T(0), tt::tolerance(eps));
```

Comment:
Are we required to type `T(0)` ? This is the question I asked you for this test `0 < fpt` in another PR